### ISSUE TITLE
[New Sensor]: Air Mentor 2S

### ISSUE THREAD

**Issue by cebrusfs** [custom-components/ble_monitor#1100]
### Sensor Description

Air Mentor 2S

### Additional information

https://www.air-mentor.com/web2017/product/air_mentor_8099af

### BLE advertisements

```shell
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x11a\x02\xb6\x00\x07\x00\n\x1b\x94\x00\x12H\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x11a\x02\xb6\x00\x07\x00\t\x1b\x94\x00\x12M\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x12Q\x03\xb9\x19\xadE(\x00\x90\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x11a\x02\xb6\x00\x07\x00\t\x1b\x94\x00\x12N\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x12Q\x03\xd5\x19\xabE(\x00\x93\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x11a\x02\xb6\x00\x06\x00\x08\x1b\x94\x00\x12O\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x12Q\x03\xc4\x19\xabE(\x00\x91\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x12Q\x03\xc4\x19\xabE(\x00\x91\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x11a\x02\xb6\x00\x06\x00\x08\x1b\x94\x00\x12P\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x12Q\x03\xca\x19\xabE(\x00\x92\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x12Q\x03\xca\x19\xabE(\x00\x92\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x11a\x02\xb6\x00\x06\x00\x08\x1b\x94\x00\x12Q\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x12Q\x03\xdf\x19\xadE(\x00\x94\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff\x11a\x02\xb6\x00\x04\x00\x05\x1b\x94\x00\x12`\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xc7\x19\xaeE(\x00\x92\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb6\x00\x04\x00\x05\x1b\x94\x00\x12a\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xc5\x19\xaeE(\x00\x92\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb7\x00\x04\x00\x05\x1b\x94\x00\x12b\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xc3\x19\xaeE(\x00\x91\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb6\x00\x04\x00\x05\x1b\x94\x00\x12c\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xda\x19\xadE(\x00\x94\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb6\x00\x03\x00\x04\x1b\x94\x00\x12d\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xcb\x19\xadE(\x00\x92\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb7\x00\x03\x00\x04\x1b\x94\x00\x12e\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xdc\x19\xadE&\x00\x94\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb7\x00\x03\x00\x04\x1b\x94\x00\x12f\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xc1\x19\xb0E(\x00\x91\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb7\x00\x03\x00\x04\x1b\x94\x00\x12g\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xce\x19\xaeE(\x00\x93\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb8\x00\x03\x00\x04\x1b\x94\x00\x12h\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xc6\x19\xadE(\x00\x92\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb8\x00\x03\x00\x04\x1b\x94\x00\x12i\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xd8\x19\xb0E(\x00\x94\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb9\x00\x03\x00\x04\x1b\x94\x00\x12j\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xcf\x19\xadE(\x00\x93\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb9\x00\x03\x00\x04\x1b\x94\x00\x12k\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xcd\x19\xb0E(\x00\x92\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb8\x00\x03\x00\x04\x1b\x94\x00\x12l\x01']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff"Q\x03\xc6\x19\xabE(\x00\x92\x05U\x00\x02']local name: 8099-AFUUID16: None,UUID128: None
INFO (Thread-2) [custom_components.ble_monitor.ble_parser] Unknown advertisement received for mac: xx:xx:xx:xx:xx:xxservice data: []manufacturer specific data: [b'\x0f\xff!a\x02\xb8\x00\x03\x00\x04\x1b\x94\x00\x12m\x01']local name: 8099-AFUUID16: None,UUID128: None
```

**Comment by cebrusfs** [comment]
Hi @Ernst79 

I saw you add https://github.com/custom-components/ble_monitor/commit/29712343aaf03fa611ea73421e6ecf8f0e433cfc for `Air Mentor pro 2` few months ago. 

I have their another device `Air Mentor 2S` which has slightly different advertisement. Now, I am working on leveraging your code and try to guess the meaning of each field by comparing the values with their mobile app. I have some work in process code at https://github.com/custom-components/ble_monitor/commit/f18aeb7fb37b5ea7618f3e83ef56ff851668e504, I could create an pull request once it is completed.

1. Could you share how you get the field meaning at `Pro 2` device? It would be help to know the format of new device's advertisement. 

2. Another question regarding humidity. In my device, it seems the sensor return the humidity (RH%) as 1 byte integer. I don't get how do you get the following fomular and calculate a more precisely value? is there any documentation that could share?

https://github.com/custom-components/ble_monitor/blob/f0ef61979d3e38e6ddf8d757cd13f200b95b4671/custom_components/ble_monitor/ble_parser/airmentor.py#L27-L29

**Comment by Ernst79** [comment]
Looks strange, I agree. In the link a gave, they use `hex_data[14:16],16`

**Comment by Ernst79** [comment]
Fixed the humi mistake in #1101
